solr/solrj/src/java/org/apache/solr/common/util/Utils.java

```diff
@@ -110,8 +110,8 @@ public static Map getDeepCopy(Map<?, ?> map, int maxDepth, boolean mutable, bool
     } else {
       copy =
           map instanceof LinkedHashMap
-              ? new LinkedHashMap<>(map.size())
-              : new HashMap<>(map.size());
+              ? CollectionUtil.newLinkedHashMap(map.size())
```

Comment:
These methods in Utils seem very related to the new CollectionUtil, and I prefer the name CollectionUtil.
Are they often used? deepCopy does not seem to be a common use case. If not, I suggest you could move them to CollectionUtil (without too much refactoring impact on other classes).

Reply:
Hmm I see your point, but there are a variety of deepCopy methods that aren't all working on collections so I think it makes sense to leave them separate from CollectionUtil.